corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -533,12 +533,15 @@ def _report_filters(self):
 
         def _make_report_filter(conf):
             col_id = self.data_source_properties[conf["property"]]['column_id']
-            return {
+            ret = {
                 "field": col_id,
                 "slug": col_id,
                 "display": conf["display_text"],
                 "type": filter_type_map[conf['format']]
             }
+            if conf['format'] == 'Date':
+                ret.update({'compare_as_string': True})
```

Comment:
I think the main arguments for doing it explicitly would be validation going into the DB and possibly performance (though I have no idea if that's a real thing).

I don't think it needs to block this.